be/src/tools/meta_tool.cpp

```diff
@@ -150,8 +152,19 @@ Status init_data_dir(const std::string& dir, std::unique_ptr<DataDir>* ret) {
         return Status::InternalError("parse root path failed");
     }
 
-    std::unique_ptr<DataDir> p(
-            new (std::nothrow) DataDir(path.path, path.capacity_bytes, path.storage_medium));
+    doris::config::tablet_map_shard_size = 256;
+    doris::config::txn_map_shard_size = 1024;
+    doris::config::txn_shard_size = 1024;
+    doris::config::partition_disk_index_lru_size = 10000;
+    doris::config::delete_bitmap_agg_cache_capacity = 104857600;
+
+    doris::ExecEnv::GetInstance()->set_cache_manager(doris::CacheManager::create_global_instance());
+    doris::ExecEnv::GetInstance()->set_dummy_lru_cache(std::make_shared<doris::DummyLRUCache>());
+
+    std::unique_ptr<doris::StorageEngine> _engine =
+            std::make_unique<doris::StorageEngine>(doris::EngineOptions {});
+    std::unique_ptr<DataDir> p(new (std::nothrow) DataDir(
```

Comment:
The lifetime of the `StorageEngine` instance constructed in this way is only within this function body, but the `DataDir` instance still needs to be used after the function returns. Therefore, the `StorageEngine` instance encounters a "use after free" issue.

Reply:
fixed